evm/utils/datatypes.py

```diff
@@ -0,0 +1,34 @@
+from evm.utils.chain import (
+    generate_vms_by_range,
+)
+
+
+class Configurable(object):
+    """
+    Base class for simple inline subclassing
+    """
+    @classmethod
+    def configure(cls,
+                  name=None,
+                  **overrides):
+
+        if name is None:
+            name = cls.__name__
+
+        if 'vms_by_range' in overrides:
+            raise ValueError("Cannot override vms_by_range.")
+
+        # Organize the Chain classes by their starting blocks.
+        if 'vm_configuration' in overrides:
+            vm_configuration = overrides['vm_configuration']
+            overrides['vms_by_range'] = generate_vms_by_range(vm_configuration)
```

Comment:
I think rather than including this in the base class since it's specific to the `VM` class, it should instead be implemented in the `VM` class using a `super` call.

```python
# in the VM class
@classmethod
def configure(cls, name, vm_configuration, **overrides):
    if 'vms_by_range' in overrides:
        raise ValueError(...)
    overrides['vms_by_range'] = generate_vms_by_range(vm_configuration)
    return super().configure(name, **overrides)
```

Reply:
@pipermerriam makes sense, though correct me if i'm wrong - shouldn't that go into `evm.chains.chain.Chain` not `evm.vm.base.VM`? if i'm wrong, i'll switch it over tomorrow

Reply:
You are correct.  I crossed wires while banging that example out.